### PR TITLE
fix(api): Sku loader

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
The algorithm for the sku dataLoader was a little bit off. This PR fixes this problem by removing some edge cases that made errors like `Could not find sku for product …` from occuring

## How to test it?
Link this PR with base.store and make sure you are able to open many different products

### CODE REVIEW

File: packages/api/src/platforms/vtex/loaders/sku.ts [hunk 1]
```diff
@@ -21,39 +21,31 @@ export const getSkuLoader = (_: Options, clients: Clients) => {
...
-    // O(n*m) sort, where n = skuIds.length and m is the number of skus per product
-    for (const product of products) {
-      const sku = product.skus.find((item) => indexById[item.id] != null)
+    const skus = skuIds.map((skuId) => {
+      const sku = skuBySkuId[skuId]
 
       if (sku == null) {
-        throw new Error(`Could not find sku for product ${product.id}`)
+        throw new Error(`Search API did not return sku with id: ${skuId}`)
       }
```
Comment (icazevedo) [inline packages/api/src/platforms/vtex/loaders/sku.ts:43]: For easier debugging, I think we should save the list of all mismatching ids and throw all the faulty ids afterward.

**Review by icazevedo — APPROVED** [review]
Code-wise, LGTM, but I think we can enhance the debugging experience.
